x/mint/depinject.go

```diff
@@ -67,28 +67,19 @@ func ProvideModule(in ModuleInputs) ModuleOutputs {
 	k := keeper.NewKeeper(
 		in.Cdc,
 		in.Environment,
-		in.StakingKeeper,
 		in.AccountKeeper,
 		in.BankKeeper,
 		feeCollectorName,
 		as,
 	)
-
-	if in.MintFn != nil && in.InflationCalculationFn != nil {
-		panic("MintFn and InflationCalculationFn cannot both be set")
-	}
-
-	// if no mintFn is provided, use the default minting function
 	if in.MintFn == nil {
```

Comment:
it should be `if in.MintFn == nil && in.StakingKeeper == nil { panic(custom minting function or staking keeper must be supplied or available) } else if in.MintFn == nil {...}`